### PR TITLE
[READY] Optional logging of message body in archive

### PR DESCRIPTION
**Pull request by stevenjack**
bang

### Problem

Currently we log the message we receive when we're archiving, this is inefficient as we've already got a reference to it and don't need it polluting the logs (This is configurable).

### Solution

Conditionally add the `message.body` to the log message if the config option depicts it.

### CODE REVIEW

File: lib/alephant/publisher/queue/sqs_helper/archiver.rb [hunk 1]
```diff
@@ -8,41 +8,73 @@ module SQSHelper
...
+          def initialize(cache, async = true, log_message_body = true)
+            @async            = async
+            @cache            = cache
+            @log_message_body = log_message_body
           end
 
           def see(message)
             return if message.nil?
-            message.tap { |m| async ? async_store(m) : store(m) }
+            async ? async_store(message) : sync_store(message)
```
Comment (darnould) [inline lib/alephant/publisher/queue/sqs_helper/archiver.rb]: No longer returning the message from `#see`, as was being tapped before. Problem?

File: lib/alephant/publisher/queue/sqs_helper/archiver.rb [hunk 2]
```diff
@@ -8,41 +8,73 @@ module SQSHelper
...
           end
 
           private
 
-          def async_store(m)
-            Thread.new { store(m) }
-            logger.metric "AsynchronouslyArchivedData"
+          def async_store(message)
+            Thread.new do
```
Comment (Integralist) [inline lib/alephant/publisher/queue/sqs_helper/archiver.rb]: Could we use http://ruby-doc.org/core-1.9.3/Thread.html#method-c-abort_on_exception-3D like http://stackoverflow.com/a/9095369/4288305 to catch Thread exceptions (so errors don't silently disappear)? Just thinking while we're editing this file

**Comment by Integralist** [comment]
:cake:

**Comment by Integralist** [comment]
@stevenjack if the specs are passing and you're confident you tested this with Spurious with success, the code itself seems ok so :cake:

File: lib/alephant/publisher/queue/sqs_helper/archiver.rb [hunk 3]
```diff
@@ -8,41 +8,75 @@ module SQSHelper
...
+            [
+              "#{self.class}#store:",
+              "'#archive/#{date_key}/#{id}'"
+            ]
+          end
+
+          def log_message(m)
+            log_message_parts(m.id).tap do |parts|
+              parts << "(#{m.body})" if log_message_body
+            end.join(" ")
```
Comment (darnould) [inline lib/alephant/publisher/queue/sqs_helper/archiver.rb]: @stevenjack If `m.body` contains newlines, they'll be emitted as part of the log, which we don't want.

Would suggest transliterating `\n` to another separator with `tr`

**Comment by darnould** [comment]
:cake:

**Pull request closed without merging**